qiskit_optimization/converters/indicator_to_inequality.py

```diff
@@ -0,0 +1,176 @@
+
+        Returns:
+            The converted problem, that contain only inequality constraints.
+
+        Raises:
+            QiskitOptimizationError: If a variable type is not supported.
+            QiskitOptimizationError: If an unsupported mode is selected.
+            QiskitOptimizationError: If an unsupported sense is specified.
+        """
+        self._src_num_var = problem.get_num_vars()
+        self._dst = QuadraticProgram(name=problem.name)
+
+        # Copy variables
+        for x in problem.variables:
+            if x.vartype == Variable.Type.BINARY:
+                self._dst.binary_var(name=x.name)
+            elif x.vartype == Variable.Type.INTEGER:
+                self._dst.integer_var(name=x.name, lowerbound=x.lowerbound, upperbound=x.upperbound)
+            elif x.vartype == Variable.Type.CONTINUOUS:
+                self._dst.continuous_var(
+                    name=x.name, lowerbound=x.lowerbound, upperbound=x.upperbound
+                )
+            else:
+                raise QiskitOptimizationError(f"Unsupported variable type {x.vartype}")
+
+        # Copy the objective function
+        constant = problem.objective.constant
+        linear = problem.objective.linear.to_dict(use_name=True)
+        quadratic = problem.objective.quadratic.to_dict(use_name=True)
+        if problem.objective.sense == QuadraticObjective.Sense.MINIMIZE:
+            self._dst.minimize(constant, linear, quadratic)
+        else:
+            self._dst.maximize(constant, linear, quadratic)
+
+        # For linear constraints
+        for l_constraint in problem.linear_constraints:
+            linear = l_constraint.linear.to_dict(use_name=True)
+            self._dst.linear_constraint(
+                linear, l_constraint.sense, l_constraint.rhs, l_constraint.name
+            )
+        # For quadratic constraints
+        for q_constraint in problem.quadratic_constraints:
+            linear = q_constraint.linear.to_dict(use_name=True)
+            quadratic = q_constraint.quadratic.to_dict(use_name=True)
+            self._dst.quadratic_constraint(
+                linear,
+                quadratic,
+                q_constraint.sense,
+                q_constraint.rhs,
+                q_constraint.name,
+            )
+        # For indicator constraints
+        for i_constraint in problem.indicator_constraints:
+            self._convert_indicator_constraint(problem, i_constraint)
+
+        return self._dst
+
+    def _convert_indicator_constraint(self, problem, indicator_const):
+        # convert indicator constraints to inequality constraints
+        new_linear = indicator_const.linear.to_dict(use_name=True)
+        new_rhs = indicator_const.rhs
+        sense = indicator_const.sense
+        new_name = indicator_const.name + self._delimiter + "indicator"
+        if sense == Constraint.Sense.LE:
+            _, lhs_ub = self._calc_linear_bounds(problem, new_linear)
+            big_m = lhs_ub - new_rhs
```

Comment:
make sure that big_m is a positive value.

Reply:
I think it does work even when big_m is a negative value in this case. When it's inactive, it becomes lhs <= the upperbounds of lhs. (When rhs is larger than the upper bound of lhs, the constraint doesn't make sense though).